### ISSUE TITLE
Touch device: writer page scrolls while dragging selection handles

### ISSUE THREAD

**Issue by nikslor** [CollaboraOnline/online#398]
**Describe the bug**
While dragging the selection handles in a writer document using the iOS app (6.4.0) the page scrolls - this is not desirable and was an issue in 4.1 too. See https://bugs.documentfoundation.org/show_bug.cgi?id=122281

**To Reproduce**
1. Open a writer document in the iOS app
2. Select some text and try to select previous/following lines too

**Expected behavior**
The page itself doesn't move and the selection over multiple lines (or pages) is easily doable.

**Actual behavior**
Because the page scrolls with the finger movement it's hard to select multiple lines.

**Tablet**
 - OS: iOS 14.1
 - Version: 6.4.0 (5)

**Additional context**
Please see https://bugs.documentfoundation.org/show_bug.cgi?id=122281 for additional issues regarding how it was solved in the app version 4.1.x

**Comment by nikslor** [comment]
Still present in 6.4.0 (14) I think this is one we should fix before marking 6.4.0 as the new stable release.

**Comment by tml1024** [comment]
I would be surprised it the iOS app is the only way this can be reproduced. Will try what happens when using Safari on iPad against a _make run_.

**Comment by tml1024** [comment]
Yep. Exactly the same effect when I connect to a _make run_ instance in Safari on iPad. If I try to tap on the selection handle and move it, it does move, but "faster" than my finger moves, and additionally, the document scrolls, too.

**Comment by tml1024** [comment]
I doubt we will be able to get this truly ideally right in a reasonable timeframe, but I wonder if it would be enough to fix it so that while you are dragging a selection handle, no scrolling of the document will happen, ever?

(Which then means that if you want to extend the selection beyond what is currently visible, you will have to pause the dragging, separately pan the document half a page or so, grab the handle again and drag further, etc. Better than the current situation?)

**Comment by mmeeks** [comment]
Hmm - doesn't sound ideal I think ... surely we need to ensure that the thresholds / areas and/or speed of page scrolling we have is nicely tuned here instead ? =)

**Comment by tml1024** [comment]
OK.

**Comment by tml1024** [comment]
Adding debug output seems to show that the root cause is that we at the same time are handling the dragging of a selection handle with L.TileLayer._onSelectionHandleDrag() **and** also for the same touch events handling panning with L.Map.TouchGesture._onPan() etc.

**Comment by mmeeks** [comment]
Nice catch =)

**Comment by nikslor** [comment]
I just tested the behavior on iOS using the testflight build 6.4.0 (16) and think this issue can be closed.

One thing that could be improved but is probably not a bug, is the following (it's a bit tricky to explain): say you have a writer document and want to select a paragraph that flows over two pages, however the second page is not yet visible on the screen. Now you start to drag the handle down the (first) page and at some point reach the end of this page. Now you drag the handle further down and the view starts to scroll down (which is good!); however the handle will not scroll with the page.

As a user I then just pack the drag handle again and as the second page is now visible I can proceed with the text selection. So it's just one minor additional step - I think we can ignore this for now.

The issue I just described might even be the root of the initial problem? Maybe the initial situation would have made it possible to scroll over two pages? Anyway: I'm in favor to keep it like it is; thanks a lot @tml1024!
